src/ussd_pmtct.js

```diff
@@ -57,42 +126,78 @@ go.app = function() {
 
         self.add("state_start", function(name) {
             self.im.user.answers = {};  // reset answers
-            return self.states.create("state_check_details");
-        });
-
-        // interstitial - checks details already saved in db
-        self.add("state_check_details", function(name) {
-            return self.states.create("state_consent");
-            // msisdn should be registered and active(?) on MomConnect...
-            /*.search_by_address({"msisdn": self.im.user.addr}, self.im, null)
-                .then(function(search) {
-                    // check whether user is already registered
-                    if (search.results.length > 0) {  // add necessary conditions to check
-                        var identity = search.results[0];
-                        console.log(identity);
-
-                        // check if lang, consent and dob already stored and set
-                        self.im.set_lang(identity.extra.language_choice);
-                        self.im.user.set_answer("consent", identity.extra.consent);
-                        self.im.user.set_answer("dob", identity.extra.dob;
-
-                        // !consent_already_given
-                        if (self.im.user.consent) {
-                            return self.states.create("state_consent", dob_already_stored);
-                        }
 
-                        // !dob_already_stored
-                        if (self.im.user.dob) {
-                            return self.states.create("state_birth_year");
-                        }
+            return self.states.create("state_check_PMTCT_subscription");
+        });
 
-                        return self.states.create("state_hiv_messages");
+        // interstitial
+        self.add("state_check_PMTCT_subscription", function(name) {
+            var msisdn = utils.normalize_msisdn(self.im.user.addr, '27');
+            self.im.user.set_answer("msisdn", msisdn);
+
+            return is.get_or_create_identity({"msisdn": msisdn})
+                .then(function(identity) {
+                    if (identity.details.pmtct) {  // on new system and PMTCT?
+                        // optout
+                        return self.states.create("state_optout_reason_menu");
+                    } else {  // register
+                        return sbm.has_active_subscription(identity.id)
+                            .then(function(active_subscription) {
+                                if (active_subscription) {
+                                    // get details (lang, consent, dob) & set answers
+                                    self.im.user.set_lang(identity.details.lang !== undefined ? identity.details.lang : "en" ); // if undefined default to english
```

Comment:
The shorthand for this is: `self.im.user.set_lang(identity.details.lang || "en")`.